crates/astria-sequencer-relayer/src/keys.rs

```diff
@@ -1,92 +1,97 @@
+use std::{
+    fs::File,
+    path::Path,
+};
+
 use bech32::{
     self,
     ToBase32,
     Variant,
 };
-use ed25519_dalek::{
-    Keypair,
-    PublicKey,
-};
-use eyre::WrapErr as _;
-use sha2::{
-    Digest,
-    Sha256,
+use ed25519_dalek::Keypair;
+use eyre::{
+    bail,
+    ensure,
+    WrapErr as _,
 };
+use serde::Deserialize;
+use tracing::instrument;
 
-const ADDRESS_LENGTH: usize = 20;
-
-/// converts an encoded private key to a ed25519_dalek::Keypair
-pub fn private_key_bytes_to_keypair(private_key: &[u8]) -> eyre::Result<Keypair> {
-    let private_key = ed25519_dalek::SecretKey::from_bytes(&private_key[..32])
-        .wrap_err("failed reading secret key from bytes")?;
-    let public_key = PublicKey::from(&private_key);
-    Ok(Keypair {
-        secret: private_key,
-        public: public_key,
-    })
+#[derive(Debug)]
+pub(crate) struct Validator {
+    pub(crate) address: Vec<u8>,
+    pub(crate) bach32_address: String,
+    pub(crate) keypair: Keypair,
 }
 
-/// converts a hex-encoded validator address to a bech32 address
-pub fn validator_hex_to_address(data: &str) -> eyre::Result<String> {
-    let address_bytes =
-        hex::decode(data).wrap_err("failed reading bytes from hex encoded string")?;
-    let address = bech32::encode("metrovalcons", address_bytes.to_base32(), Variant::Bech32)
-        .wrap_err("failed converting bytes to bech32 address")?;
-    Ok(address)
-}
+impl Validator {
+    #[instrument(skip_all, fields(path = %path.as_ref().display(), err))]
+    pub(crate) fn from_path(path: impl AsRef<Path>) -> eyre::Result<Self> {
+        use base64_serde::base64_serde_type;
+        base64_serde_type!(Base64Standard, base64::engine::general_purpose::STANDARD);
+
+        #[derive(Deserialize, zeroize::Zeroize, zeroize::ZeroizeOnDrop)]
+        pub struct Key {
+            #[serde(deserialize_with = "Base64Standard::deserialize")]
+            value: Vec<u8>,
+        }
+
+        #[derive(Deserialize)]
+        struct JsonRepr {
+            #[serde(with = "hex")]
+            address: Vec<u8>,
+            pub_key: Key,
+            priv_key: Key,
+        }
+
+        let f = File::open(&path).wrap_err("failed to open validator key file")?;
+        let Ok(read_keys): Result<JsonRepr, _> = serde_json::from_reader(f) else {
+            // silently drop the error to ensure the deserializer never leaks the keys
+            // now or in the future.
+            bail!("failed to read the validator key from file");
+        };
+        let keypair = Keypair::from_bytes(&read_keys.priv_key.value)
+            .wrap_err("failed generating ed25519 dalek keypair from validator private key")?;
```

Comment:
```suggestion
            .wrap_err("failed generating ed25519 keypair from validator private key")?;
```

i think the dalek part is just naming they gave the crate :p